tests/services/test_service.py

```diff
@@ -70,3 +71,99 @@ def test_simple_flow(app, consumer, service, identity_simple, input_data):
     # - search
     res = service.search(identity_simple, q=f"id:{id_}", size=25, page=1)
     assert res.total == 0
+
+
+def test_read_all_cache(app, es_clear, service, identity_simple, input_data):
+    # FIXME: https://github.com/inveniosoftware/pytest-invenio/issues/30
+    current_cache.clear()
+    # Create an items
+    item_one = service.create(identity_simple, input_data)
+    item_two = service.create(identity_simple, input_data)
+    Record.index.refresh()
+
+    records = service.read_all(
+        identity_simple, fields=["metadata.title"])  # cached by default
+
+    assert records.total == 2
+    for record in records.hits:
+        assert record["id"] is not None
+        assert list(record["metadata"].keys()) == ["title"]
+        assert record["metadata"]["title"] == "Test"
+
+    cached = current_cache.get("metadata.title")
```

Comment:
Minor: Here I would call again ``read_all()`` to ensure that it works. I think you can just assert that there's a cache key in the cache without testing what is being cached.

Reply:
Fair, I wanted to sort of make sure that the cache was correct... but is true that its gonna be the same that gets returned...